### DISCUSSION TITLE
How to configure mutate webhook for vault mtls

### DISCUSSION THREAD

**Discussion by Renz2018** [bank-vaults/bank-vaults#2308] [Q&A] [answered]
The Vault(runs outside Kubernetes.) server has enabled TLS mutual authentication in addition to Kubernetes authentication. How should it be configured mutate webhook?
Like vault agent injector:
```
[vault.hashicorp.com/client-cert](https://developer.hashicorp.com/vault/docs/platform/k8s/injector/annotations#vault-hashicorp-com-client-cert) - path of the client certificate used when communicating with Vault via mTLS.

[vault.hashicorp.com/client-key](https://developer.hashicorp.com/vault/docs/platform/k8s/injector/annotations#vault-hashicorp-com-client-key) - path of the client public key used when communicating with Vault via mTLS.
```

**Comment by ramizpolic** [comment, marked as answer]
Currently [TLS Certificate](https://developer.hashicorp.com/vault/docs/auth/cert) authentication is not supported. We support a selected number of auth methods listed [here](https://github.com/bank-vaults/vault-sdk/blob/7090d894d587e7943a6f6a6809c260679602bac1/vault/client.go#L157-L186). This would be a great way to help and contribute to the project as well if this is something that would be useful for you. I am here to assist.

**Reply by Renz2018** [reply]
thanks!!!
